bot/cogs/utils.py

```diff
@@ -0,0 +1,82 @@
+import logging
+from email.parser import HeaderParser
+from io import StringIO
+
+
+from discord import Colour, Embed
+from discord.ext.commands import AutoShardedBot, Context, command
+
+from bot.constants import Roles
+from bot.decorators import with_role
+
+log = logging.getLogger(__name__)
+
+
+class Utils:
+    """
+    A selection of utilities which don't have a clear category.
+    """
+
+    def __init__(self, bot: AutoShardedBot):
+        self.bot = bot
+
+        self.base_pep_url = "http://www.python.org/dev/peps/pep-"
+        self.base_github_pep_url = "https://raw.githubusercontent.com/python/peps/master/pep-"
+
+    @command(name="pep()", aliases=["pep", "get_pep"])
+    @with_role(Roles.verified)
+    async def pep_search(self, ctx: Context, pep_number: str):
+        """
+        Fetches information about a PEP and sends it to the user
+        """
+
+        # Attempt to fetch the PEP from Github.
+        pep_url = f"{self.base_github_pep_url}{pep_number.zfill(4)}.txt"
```

Comment:
you can do zfill with fstrings too. i'd personally do `f"{self.base_github_pep_url}{pep_number:04}.txt"`.